Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1193,12 +1214,33 @@ void RenderBlockFlow::determineLogicalLeftPositionForChild(RenderBox& child, App
     // If the child is being centred then the margin calculated to do that has factored in any offset required to
     // avoid floats, so use it if necessary.
 
-    if (style().textAlign() == TextAlignMode::WebKitCenter || child.style().marginStartUsing(&style()).isAuto())
+    if (style().textAlign() == TextAlignMode::WebKitCenter || child.style().marginStartUsing(&style()).isAuto()) {
         newPosition = std::max(newPosition, positionToAvoidFloats + childMarginStart);
-    else if (positionToAvoidFloats > initialStartPosition)
+        setLogicalLeftForChild(child, style().isLeftToRightDirection() ? newPosition : totalAvailableLogicalWidth - newPosition - logicalWidthForChild(child), applyDelta);
+        return;
+    }
+
+    if (positionToAvoidFloats > initialStartPosition)
         newPosition = std::max(newPosition, positionToAvoidFloats);
+    newPosition = computeSelfAlignedLogicalLeftPositionForChild(newPosition);
+    setLogicalLeftForChild(child, newPosition, applyDelta);
+}
 
-    setLogicalLeftForChild(child, style().isLeftToRightDirection() ? newPosition : totalAvailableLogicalWidth - newPosition - logicalWidthForChild(child), applyDelta);
+void RenderBlockFlow::applyStretchAlignmentToChildIfNeeded(RenderBox& child)
+{
+    const auto hasAutoLogicalWidth = isHorizontalWritingMode() ? child.style().width().isAuto() : child.style().height().isAuto();
+    const auto hasAutoLogicalInlineMargins = [&] {
+        if (isHorizontalWritingMode())
+            return child.style().marginLeft().isAuto() || child.style().marginRight().isAuto();
+        return child.style().marginTop().isAuto() || child.style().marginBottom().isAuto();
+    }();
+    const auto childSelfAlignmentStyle = child.style().resolvedJustifySelf(&style(), ItemPosition::Normal);
+    const auto allowedToStretchChild = childSelfAlignmentStyle.position() == ItemPosition::Stretch && hasAutoLogicalWidth && !hasAutoLogicalInlineMargins;
+    if (!allowedToStretchChild)
+        return;
+    const auto stretchedLogicalWidth = availableLogicalWidth();
+    const auto desiredLogicalWidth = child.constrainLogicalWidthInFragmentByMinMax(stretchedLogicalWidth, contentWidth(), *this, nullptr);
+    child.setLogicalWidth(desiredLogicalWidth);
```

Comment:
while I can't point you to a specific rule in the style guideline, the heavy use of const auto here is somewhat surprising (personally not against const).

Reply:
I just const-by-default since I think it's easier to reason about among other things, I know the rest of the codebase doesn't really do this though. I can probably keep this?

Reply:
(again, while I support const) I think for the sake of consistency, they should be dropped.